pdb2sql/pdb2sqlcore.py

```diff
@@ -228,63 +242,72 @@ def _fix_chainID(self):
 
     def get_colnames(self):
         cd = self.conn.execute('select * from atom')
-        print('Possible column names are:')
         names = list(map(lambda x: x[0], cd.description))
-        print('\trowID')
+        names = ['rowID'] + names
+        return names
+
+    def print_colnames(self):
+        names = self.get_colnames()
+        print('Possible column names are:')
         for n in names:
             print('\t' + n)
 
     # print the database
-    def prettyprint(self):
+    def pprint(self):
         import pandas.io.sql as psql
         df = psql.read_sql("SELECT * FROM ATOM", self.conn)
         print(df)
 
-    def uglyprint(self):
+    def print(self):
         ctmp = self.conn.cursor()
         ctmp.execute("SELECT * FROM ATOM")
         print(ctmp.fetchall())
 
     # get the properties
-    def get(self, atnames, **kwargs):
-        '''Exectute a simple SQL query that extracts values of attributes for certain condition.
+    def get(self, columns, **kwargs):
+        '''Exectute simple SQL query to extract values of attributes
+            for certain conditions.
 
         Args :
-            attribute (str) : attribute to retreive eg : ['x','y,'z'], 'xyz', 'resSeq'
-                              if None all the attributes are returned
-
-            **kwargs : argument to select atoms eg : name = ['CA','O'], chainID = 'A', no_name = ['H']
+            columns (str): columns to retreive, eg: "x,y,z".
+                if "*" all the columns are returned.
+                Check all available columns by print_colnames().
+            **kwargs: argument to select atoms, dict value must be list.
+                eg: name = ['CA', 'O'], chainID = ['A'],
+                or no_name = ['CA', 'C'], no_chainID = ['A'].
 
         Returns:
-            data : array containing the value of the attributes
+            data: list containing the value of the attributes
 
-        Example :
-        >>> db.get('x,y,z',chainID='A',no_name=['H']")
+        Examples:
+            >>> db.get('x,y,z', chainID=['A'], no_resName=['ALA', 'TRP'])
         '''
+        # check arguments format
+        valid_colnames = self.get_colnames()
+
+        if not isinstance(columns, str):
+            raise TypeError("argument columns must be str")
+
+        if columns != '*':
+            for i in columns.split(','):
+                if i.strip() not in valid_colnames:
+                    raise ValueError(
+                        f'Invalid column name {i}. Possible names are\n'
+                        f'{self.get_colnames()}')
 
         # the asked keys
         keys = kwargs.keys()
 
-        # check if the column exists
```

Comment:
why did you remove that?

Reply:
It is not removed but refactored. The former code queries SQL with one column per time, and the new code queries all available columns and then check the existence.